src/wkwebview/class/wry_web_view.rs

```diff
@@ -0,0 +1,152 @@
+// Copyright 2020-2024 Tauri Programme within The Commons Conservancy
+// SPDX-License-Identifier: Apache-2.0
+// SPDX-License-Identifier: MIT
+
+use std::collections::HashMap;
+
+use objc2::{
+  declare_class,
+  mutability::MainThreadOnly,
+  rc::Retained,
+  runtime::{Bool, MessageReceiver, ProtocolObject},
+  ClassType, DeclaredClass,
+};
+use objc2_app_kit::NSEvent;
+use objc2_foundation::NSUUID;
+use objc2_web_kit::WKWebView;
+
+use crate::{
+  wkwebview::{drag_drop, synthetic_mouse_events},
+  DragDropEvent,
+};
+
+pub struct WryWebViewIvars {
+  pub(crate) is_child: bool,
+  #[cfg(target_os = "macos")]
+  pub(crate) drag_drop_handler: Box<dyn Fn(DragDropEvent) -> bool>,
+  #[cfg(target_os = "macos")]
+  pub(crate) accept_first_mouse: objc2::runtime::Bool,
+  pub(crate) custom_protocol_task_ids: HashMap<usize, Retained<NSUUID>>,
+}
+
+declare_class!(
+  pub struct WryWebView;
+
+  unsafe impl ClassType for WryWebView {
+    type Super = WKWebView;
+    type Mutability = MainThreadOnly;
+    const NAME: &'static str = "WryWebView";
+  }
+
+  impl DeclaredClass for WryWebView {
+    type Ivars = WryWebViewIvars;
+  }
+
+  unsafe impl WryWebView {
+    #[method(performKeyEquivalent:)]
+    fn perform_key_equivalent(
+      &self,
+      _event: &NSEvent,
+    ) -> Bool {
+      // This is a temporary workaround for https://github.com/tauri-apps/tauri/issues/9426
+      // FIXME: When the webview is a child webview, performKeyEquivalent always return YES
+      // and stop propagating the event to the window, hence the menu shortcut won't be
+      // triggered. However, overriding this method also means the cmd+key event won't be
+      // handled in webview, which means the key cannot be listened by JavaScript.
+      if self.ivars().is_child {
+        Bool::NO
+      } else {
+        unsafe {
+          let result: Bool = self.send_super_message(
+            WKWebView::class(),
+            objc2::sel!(performKeyEquivalent:),
+            (_event,),
+          );
+          result
+        }
+      }
+    }
+
+    #[method(acceptsFirstMouse:)]
+    fn accept_first_mouse(
+      &self,
+      _event: &NSEvent,
+    ) -> Bool {
+        self.ivars().accept_first_mouse
+    }
+  }
+
+  // Drag & Drop
+  #[cfg(target_os = "macos")]
+  unsafe impl WryWebView {
```

Comment:
Nit: Implement the protocol for this.
```suggestion
  unsafe impl NSDraggingDestination for WryWebView {
```